plugin-server/src/cdp/services/native-destination-executor.service.ts

```diff
@@ -0,0 +1,257 @@
+import { Histogram } from 'prom-client'
+
+import { PluginsServerConfig } from '~/types'
+import { parseJSON } from '~/utils/json-parse'
+
+import { logger } from '../../utils/logger'
+import { fetch, FetchOptions, FetchResponse } from '../../utils/request'
+import { tryCatch } from '../../utils/try-catch'
+import { NATIVE_HOG_FUNCTIONS_BY_ID } from '../templates'
+import { Response } from '../templates/types'
+import { CyclotronJobInvocationHogFunction, CyclotronJobInvocationResult } from '../types'
+import { CDP_TEST_ID, createAddLogFunction, isNativeHogFunction } from '../utils'
+import { createInvocationResult } from '../utils/invocation-utils'
+import { getNextRetryTime, isFetchResponseRetriable } from './hog-executor.service'
+
+const nativeDestinationExecutionDuration = new Histogram({
+    name: 'cdp_native_execution_duration_ms',
+    help: 'Processing time and success status of native plugins',
+    // We have a timeout so we don't need to worry about much more than that
+    buckets: [0, 10, 20, 50, 100, 200],
+})
+
+class FetchError extends Error {
+    constructor(message?: string) {
+        super(message)
+    }
+}
+
+const convertFetchResponse = (response: FetchResponse, text: string): Response => {
```

Comment:
Feels like you've basically copy-pasted the segment executor which feels nasty. We don't need to convert to these types when we could just be handing our fetch function and types in directly. Lots of code here is basically mapping to a "segment" style executor which we don't have to conform to. 

Was there some specific design thinking you had here that I'm missing?